### PR TITLE
Ddpb 2223 casrec report csv

### PR DESCRIPTION
**Pull request by slizzio**

### CODE REVIEW

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php [hunk 1]
```diff
@@ -110,4 +110,79 @@ public function findDownloadableOlderThan(\DateTime $olderThan, $limit)
...
+
+        // to filter out multiple submissions, look at the rs.createdOn as being greater or equal to the original
+        // report submission date
+        $qbSelect->andWhere('rs.createdOn >= r.submitDate OR rs.createdOn >= ndr.submitDate');
+        $qbSelect->andWhere('r.submitted = true OR ndr.submitted = true');
+        $qbSelect->andWhere('r.submitDate IS NOT NULL OR ndr.submitDate IS NOT NULL');
+        $qbSelect
+            ->orderBy('rs.' . $orderBy, $order)
+            ->setFirstResult($offset);
+            //->setMaxResults($limit);
```
Comment (elvisciotti) [inline src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: to clean

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php [hunk 2]
```diff
@@ -110,4 +110,79 @@ public function findDownloadableOlderThan(\DateTime $olderThan, $limit)
...
+        $this->_em->getFilters()->getFilter('softdeleteable')->disableForEntity(User::class); //disable softdelete for createdBy, needed from admin area
```
Comment (elvisciotti) [inline src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: I removed the softdelete for users some time ago I think

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php [hunk 3]
```diff
@@ -110,4 +110,79 @@ public function findDownloadableOlderThan(\DateTime $olderThan, $limit)
...
+            'records'=>$records,
+        ];
+    }
+
+    /**
+     * Calculate FromDate for ReportSubmissions. Used for CSV generation to include weekends reports on Monday.
+     *
+     * @return \DateTime
+     */
+    private function calculateFromDate()
```
Comment (elvisciotti) [inline src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: does it need to renamed maybe ?

File: src/AppBundle/Controller/Report/ReportSubmissionController.php [hunk 4]
```diff
@@ -136,4 +138,45 @@ public function setUndownloadable($id, Request $request)
...
+
+        $this->setJmsSerialiserGroups(
+            [
+                'report-submission',
+                'report-type',
+                'report-client',
+                'ndr-client',
+                'ndr',
+                'report',
+//                'report-period',
```
[the remark this answers is not in the document]
Comment (slizzio) [reply src/AppBundle/Controller/Report/ReportSubmissionController.php]: removed

File: src/AppBundle/Entity/Report/ReportSubmission.php [hunk 5]
```diff
@@ -17,6 +17,7 @@
 class ReportSubmission
 {
     const REMOVE_FILES_WHEN_OLDER_THAN = '-30 days';
+    const STATS_FILE_PATH = '/tmp/dd-all-submissions-stats.csv';
```
[the remark this answers is not in the document]
Comment (slizzio) [reply src/AppBundle/Entity/Report/ReportSubmission.php]: yep good spot, part of old solution

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php (hunk 1 above)
Comment (slizzio) [reply src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: removed, thought we might have needed a limit, but unlikely now.

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php (hunk 2 above)
Comment (slizzio) [reply src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: Yes, just checked, removed this. I copied it from another method in the class. Probably needs removing from their too although shouldn't do any harm

File: src/AppBundle/Entity/Repository/ReportSubmissionRepository.php (hunk 3 above)
Comment (slizzio) [reply src/AppBundle/Entity/Repository/ReportSubmissionRepository.php]: hmm yes.. agreed... determineCreationFromDate? its a private method to ensure that we get all report submissions from midnight the previous day, unless its Monday in which case we return the entire weekend's report submissions.

**Pull request merged**